controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -56,8 +56,7 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, contro
 		return nil
 	}
 
-	machinesWithHealthyAPIServer := controlPlane.Machines.Filter(collections.HealthyAPIServer())
```

Comment:
I think HealthyAPIServer is now unused, let's delete it

Reply:
Done, question: dropping this: should we remove that "drop" from the cherry-pick

Reply:
The latter